First/FightSimulation/EloFightSimulator.cs

```diff
@@ -22,11 +22,16 @@ public FightOutcome SimulateFight(Main.Fight fight)
                 // fighter 1 won
                 updateElo(fight.b1, fight.b2);
                 winner = fight.b1;
+                fight.b1.record.Wins++;
+                fight.b2.record.Losses++;
+
             }
             else
             {
                 updateElo(fight.b2, fight.b1);
                 winner = fight.b2;
+                fight.b2.record.Wins++;
```

Comment:
The fight simulator should not update elos or fight record. 
Fight sim should return result of the fight and then the result should be used to update the elo (by passing in result and current elos to Elo Calculator class?). Similarly fight records of the fighters should not be touched by sim class. Fight and Fighter should prob be treated as constants by FightSim/Elo classes so we dont intertwine the classes too much

Reply:
I don't know why we need another class to update the results of the calculation. A simulator class simulates the fight then at the end of SimulateFight() method, the results needs to be updated (hence the name).

Reply:
what if i write my own fight simulator? will i then need to put the updateElo() in mine as well - seems like duplication of code. I think its fine either way as this class is more for testing but in the future it would be good to have a seperate class for calculating elo that we can call in any context

Reply:
You would need to update rank of the fighter somehow (you may have  different formula) and other fight-related outcomes (number of wins, etc.)